src/components/ErrorCard.tsx

```diff
@@ -0,0 +1,26 @@
+import { Dispatch, SetStateAction } from "react";
+
+type ErrorCard = {
+  error: Error | null;
+  setError: any
+}
+
+export default function ErrorCard({ error, setError }:ErrorCard) {
+  if (error) {
+    setTimeout(() => {
+      setError(null);
+    }, 4000);
+  }
+  return (
+    <>
+      {error ? (
+        <div
+          className="p-4 mb-4 text-sm text-red-700 bg-red-100 rounded-lg"
+          role="alert"
+        >
+          {error.toString()}
+        </div>
+      ) : null}
+    </>
```

Comment:
Lets rely on error to be defined instead. So the component is not rendered with empty content if theres no error. This way we save us a DOM operation ourselves!